spec/factories/digital_objects/items.rb

```diff
@@ -23,5 +23,11 @@
         digital_object.primary_project = create(:project)
       end
     end
+
+    trait :with_asset do
+      after(:create) do |digital_object|
+        digital_object.append_child_uid(create(:asset, parent: digital_object).uid)
```

Comment:
One concern about this method of adding an asset to the item factory object is that you've set up a single-directional relationship between these two objects when normally there would be a bidirectional relationship.  Could lead to inconsistencies during testing.